packages/react-checkbox/Spec.md

```diff
@@ -2,62 +2,267 @@
+- A Checkbox has two size variants: `medium (default)` and `large`.
+- A Checkbox has three appearance variants: `disabled`, `unchecked`, `checked`, and `indeterminate`.
+- A Checkbox has a `circular` variant.
 
 ## API
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+### Checkbox Props
+
+```tsx
+/**
+ * Checkbox Props
+ */
+export interface CheckboxProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
+  /**
+   * Label to be rendered with the checkbox.
+   */
+  label?: ShorthandProps<LabelProps>;
+
+  /**
+   * Icon to be displayed when the checkbox is in the checked state.
+   */
+  icon?: React.ReactElement;
+
+  /**
+   * Disabled state of the checkbox.
+   */
+  disabled?: boolean;
+
+  /**
+   * A checkbox can be rendered with a circular shape.
+   */
+  circular?: boolean;
+
+  /**
+   * A checkbox's state can be controlled.
+   * @defaultvalue false
+   */
+  checked?: 'indeterminate' | boolean;
+
+  /**
+   * Whether the checkbox should be rendered as checked by default.
+   */
+  defaultChecked?: 'indeterminate' | boolean;
+
+  /**
+   * Checkbox supports two different checkbox sizes.
+   * @defaultvalue 'medium'
+   */
+  size?: 'medium' | 'large';
+
+  /**
+   * Determines whether the label should be positioned before (start) or after (end) the checkbox.
+   * @defaultvalue 'end'
+   */
+  labelPosition?: 'start' | 'end';
+
+  /**
+   * ID of the root element that wraps the checkbox and label.
+   */
+  rootId?: string;
+
+  /**
+   * ID of the <input/> element that represents the checkbox.
+   */
+  id?: string;
+
+  /**
+   * Callback to be called when the checked state value changes.
+   */
+  onChange?: (ev?: React.FormEvent<HTMLElement | HTMLInputElement>, checked?: CheckboxOnChangeData) => void;
+}
+
+/**
+ * Data for the onChange event for checkbox.
+ */
+export interface CheckboxOnChangeData {
+  checked?: 'indeterminate' | boolean;
+}
+```
+
+Some feedback needed for props are:
+
+- `checked` will work similar to `aria-checked` and have `true`, `false`, and `indeterminate` options.
+- A `rootId` is needed since the input and label will share an id and `rootId` will be used for the root element.
+- To initially make the checkbox simple, the icon will be a `ReactElement` instead of a slot.
 
 ## Structure
 
-- _**Public**_
-- _**Internal**_
-- _**DOM** - how the component will be rendered as HTML elements_
+### **Public**
 
-## Migration
+```tsx
+<Checkbox label="Example Checkbox" />
+```
 
-_Describe what will need to be done to upgrade from the existing implementations:_
+### **Internal**
 
-- _Migration from v8_
-- _Migration from v0_
+```tsx
+<slots.root {...slotProps.root}>
+  {state.labelPosition === 'start' && <slots.label {...slotProps.label} />}
+  <div className={state.checkboxClassName}>
+    <div className={state.iconClassName}>{icon}</div>
+    <input
```

Comment:
IMO `input` should be also a slot as otherwise there is no a way for customers to pass props to `<input />` element.